decompiler/frontend/binaryninja/handlers/globals.py

```diff
@@ -42,7 +42,7 @@ def lift_global_variable(self, variable: DataVariable, view: BinaryView,
                     self._lifter.lift(variable.type),
                     ssa_label=parent.ssa_memory_version if parent else 0,
                     initial_value=self._lifter.lift(view.get_data_var_at(variable.value), view=view) if isinstance(variable.type, PointerType) \
-                    and variable.value != 0 else Constant(variable.value) # pointer can point to NULL as well
+                    and variable.value != 0 and variable.address != variable.value else Constant(variable.value) # pointer can point to NULL/itself 
```

Comment:
I'm somewhat of a tenary operator enjoyer as well but spanning two whole lines with **3** conditions may be a little too much for good readablity.

Could we extract a function `get_initial_value`? As far as I can see, is would only require the parameter `variable`.

Reply:
Done